src/core/chart-container.tsx

```diff
@@ -85,6 +87,7 @@ export function ChartContainer({
           >
             {verticalAxisTitle}
             {chart(effectiveChartHeight)}
+            {noData}
```

Comment:
The noData placement now depends on the legend placement. That is to ensure the no-data message is correctly centered.

Before:
<img width="1362" height="471" alt="Image" src="https://github.com/user-attachments/assets/ce113349-31fd-46b7-81d6-5c5977965bc9" />

After:
<img width="1362" height="471" alt="Image" src="https://github.com/user-attachments/assets/3aa8d5c7-19ae-4e6f-895e-7689128b80a3" />